code/datums/black_market_item.dm

```diff
@@ -72,62 +74,118 @@ var/list/black_market_items = list()
+	spawn(rand(150,450))
+		var/obj/item = spawn_item(get_turf(user), radio, user)
+		if(!item)
 			return 0
-		after_spawn(I,user)
-		var/icon/tempimage = icon(I.icon, I.icon_state)
-		end_icons += tempimage
-		var/tempstate = end_icons.len
-
-		if(ishuman(user))
-			var/mob/living/carbon/human/A = user
-
-			if(istype(I, /obj/item))
-				A.put_in_any_hand_if_possible(I)
-
-			U.purchase_log += {"[user] ([user.ckey]) bought <img src="logo_[tempstate].png"> [name] for [get_cost()]."}
-			//stat_collection.black_market_purchase(src, I, user)
-			if(get_stock() != 0)
-				round_stock -= 1
+		after_spawn(item,1,user)
+		item.ThrowAtStation(30,2,direction)
+		if(get_stock() != 0)
+			round_stock -= 1	
+		spawn(rand(300,600))
+			if(!radio.nanotrasen_variant && prob(sps_chances[1]))
+				SPS_black_market_alert(src, "The SPS decryption complex has detected an illegal black market purchase of item [name]. It was launched at the station recently.")
+	
+/datum/black_market_item/proc/spawn_normal(var/obj/item/device/illegalradio/radio, var/mob/user)
+	var/list/potential_locations = list()
+	var/turf/spawnloc = null
+	for(var/area/maintenance/A in areas)
+		if(A != /area/maintenance/atmos)
+			potential_locations.Add(A)
+	var/area/selected_area 
+	while(potential_locations.len)
+		selected_area = pick(potential_locations)
+		potential_locations.Remove(selected_area)
+		for(var/turf/simulated/floor/floor in selected_area.contents)
+			if(!floor.has_dense_content())
+				spawnloc = floor
+				break	
+	if(!spawnloc)
+		return
+	var/time_to_spawn = rand(300,600)
+	log_transaction("The item was teleported to the [selected_area.name].", user)
+	radio.visible_message("The uplink beeps: <span class='warning'>Your item has been sent through bluespace. It will appear somewhere in [selected_area.name] in [time_to_spawn/10] seconds.</span>")
+	spawn(time_to_spawn)
+		var/obj/item = spawn_item(spawnloc, radio, user)
+		after_spawn(item,2,user)
+		if(get_stock() != 0)
+			round_stock -= 1
+		spawn(rand(300,600))
+			if(!radio.nanotrasen_variant && prob(sps_chances[2]))
+				SPS_black_market_alert(src, "The SPS decryption complex has detected an illegal black market purchase of item [name]. It was teleported to your station's maintenance recently.")
+		
+/datum/black_market_item/proc/spawn_expensive(var/obj/item/device/illegalradio/radio, var/mob/user)
+	var/obj/item = spawn_item(get_turf(user), radio, user)
+	if(!item)
+		return 0
+	after_spawn(item,3,user)
 
-				
-		user.set_machine(U)
-		U.interact(user)
+	if(ishuman(user))
+		var/mob/living/carbon/human/A = user
+		if(istype(item, /obj/item))
+			A.put_in_any_hand_if_possible(item)
+		if(get_stock() != 0)
+			round_stock -= 1
+	log_transaction("The item was teleported directly to him.", user)
+	if(prob(99))
+		radio.visible_message("The uplink beeps: <span class='warning'>Thank you for your purchase!</span>")	
```

Comment:
formatted, this looks really weird - uniform formatting for whole message, and consider a bicon for the radio at the start of the message

is it intended that anyone nearby will see the visible message when you buy something?

Reply:
Yes, it is intended. This is how SPS messages look, so I think it looks fine. Might add a bicon tho.

Reply:
After testing it out I think it looks fine. I changed the name to be [name] instead of uplink, which I think looks better, but the bicon looks ugly. See: SPS, since the sprite is so small the bicon is basically a pixel wide.